### ISSUE TITLE
DOI registration: wrong `g.identity` in celery task:

### ISSUE THREAD

**Issue by ppanero** [inveniosoftware/invenio-rdm-records#964]
```
  File "/Users/nico/.local/share/virtualenvs/my-site-5PKhBfd9-python/lib/python3.8/site-packages/invenio_rdm_records/resources/serializers/datacite/schema.py", line 184, in get_type
    props = get_vocabulary_props(
  File "/Users/nico/.local/share/virtualenvs/my-site-5PKhBfd9-python/lib/python3.8/site-packages/invenio_rdm_records/resources/serializers/utils.py", line 23, in get_vocabulary_props
    g.identity,
  File "/Users/nico/.local/share/virtualenvs/my-site-5PKhBfd9-python/lib/python3.8/site-packages/flask/ctx.py", line 51, in __getattr__
    raise AttributeError(name) from None
AttributeError: identity
```

**Comment by ppanero** [comment]
TL;DR/TODO: revert https://github.com/inveniosoftware/invenio-rdm-records/pull/959
